packages/arb-validator-core/ethbridge/arbClient.go

```diff
@@ -190,30 +194,97 @@ type TransactAuth struct {
 	auth *bind.TransactOpts
 }
 
-func (t *TransactAuth) makeTx(ctx context.Context, txFunc func(auth *bind.TransactOpts) (*types.Transaction, error)) (*types.Transaction, error) {
-	const smallNonceRepeatCount = 5
-	const smallNonceError = "Try increasing the gas price or incrementing the nonce."
+func (t *TransactAuth) makeContract(ctx context.Context, contractFunc func(auth *bind.TransactOpts) (ethcommon.Address, *types.Transaction, interface{}, error)) (ethcommon.Address, *types.Transaction, error) {
+	auth := t.getAuth(ctx)
+
+	addr, tx, _, err := contractFunc(auth)
+
+	if auth.Nonce == nil {
+		// Not incrementing nonce, so nothing else to do
+		if err != nil {
+			log.Err(err).Str("nonce", "nil").Msg("makeContract")
+			return addr, nil, err
+		}
+
+		txJSON, err := tx.MarshalJSON()
+		if err != nil {
+			log.Err(err).Str("nonce", "nil").Msg("makeContract failed to marshal tx into json")
+			return addr, tx, err
+		}
 
+		log.Info().RawJSON("tx", txJSON).Str("nonce", "nil").Str("sender", t.auth.From.Hex()).Msg("makeContract")
+		return addr, nil, err
+	}
+
+	for i := 0; i < smallNonceRepeatCount && err != nil && strings.Contains(err.Error(), smallNonceError); i++ {
+		// Increment nonce and try again
+		log.Err(err).Str("nonce", auth.Nonce.Text(16)).Msg("makeContract incrementing nonce and submitting tx again")
+
+		auth.Nonce = t.auth.Nonce.Add(t.auth.Nonce, big.NewInt(1))
```

Comment:
On the left side you have `auth.Nonce` and the right `t.auth.Nonce`. I'm guessing these should actually be the same

Reply:
nope, this is actually on purpose!  `t.auth.Nonce.Add` updates t.auth.Nonce, but I also want to update `auth.Nonce` separately, so this is a convenient way of doing it.